mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenPropertyResolver.java

```diff
@@ -1,32 +1,97 @@
 package org.arquillian.smart.testing.mvn.ext;
 
 import java.util.Arrays;
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+import static org.arquillian.smart.testing.mvn.ext.ApplicablePlugins.FAILSAFE;
+import static org.arquillian.smart.testing.mvn.ext.ApplicablePlugins.SUREFIRE;
 
 class MavenPropertyResolver {
 
+    private final Model model;
+    private final Plugin surefirePlugin;
+    private final Plugin failsafePlugin;
+
     private static final Pattern TEST_CLASS_PATTERN = Pattern.compile("[^a-z0-9 ]", Pattern.CASE_INSENSITIVE);
 
+    MavenPropertyResolver(Model model) {
+        this.model = model;
+        this.surefirePlugin = getPlugin(SUREFIRE);
+        this.failsafePlugin = getPlugin(FAILSAFE);
+    }
+
     static boolean isSkipTestExecutionSet() {
         return isSkipTests() || isSkip();
     }
 
-    static private boolean isSkipTests() {
-        return Boolean.valueOf(System.getProperty("skipTests", "false"));
+    boolean isSkipITs() {
+        return Boolean.valueOf(System.getProperty("skipITs", "false")) || isSkipITsSetInPom();
+    }
+
+    boolean isSkipTestsSetInPom() {
+        if (surefirePlugin != null && failsafePlugin == null) {
+            Xpp3Dom surefirePluginConfiguration = (Xpp3Dom) surefirePlugin.getConfiguration();
+            return isPropertySetInPluginConfiguration(surefirePluginConfiguration, "skip") ||
+                isPropertySetInPluginConfiguration(surefirePluginConfiguration, "skipTests");
+        }
+        return isPropertyInPom("maven.test.skip") || isPropertyInPom("skipTests");
+    }
+
+    static boolean isSpecificTestClassSet() {
+        String testClasses = System.getProperty("test");
+        return testClasses != null && !containsPattern(testClasses);
     }
 
-    static boolean isSkipITs() {
-        return Boolean.valueOf(System.getProperty("skipITs", "false"));
+    static private boolean isSkipTests() {
```

Comment:
It'll be good to follow java specification for [this].(https://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.1.1)
`private static boolean isSkipTests()`